streams-bootstrap-core/src/main/java/com/bakdata/kafka/consumer/ConsumerCleanUpRunner.java

```diff
@@ -0,0 +1,169 @@
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Clean up all topics specified by a {@link ConsumerTopicConfig}
+ */
+@Slf4j
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ConsumerCleanUpRunner implements CleanUpRunner {
+    private final @NonNull ConsumerTopicConfig topics;
+    private final @NonNull Map<String, Object> kafkaProperties;
+    private final @NonNull String groupId;
+    private final @NonNull ConsumerCleanUpConfiguration cleanHooks;
+
+    /**
+     * Create a new {@code ConsumerCleanUpRunner} with default {@link ConsumerCleanUpConfiguration}
+     *
+     * @param topics topic configuration
+     * @param kafkaProperties configuration to connect to Kafka admin tools
+     * @param groupId consumer group id to clean up
+     * @return {@code ConsumerCleanUpRunner}
+     */
+    public static ConsumerCleanUpRunner create(@NonNull final ConsumerTopicConfig topics,
+            @NonNull final Map<String, Object> kafkaProperties,
+            @NonNull final String groupId) {
+        return create(topics, kafkaProperties, groupId, new ConsumerCleanUpConfiguration());
+    }
+
+    /**
+     * Create a new {@code ConsumerCleanUpRunner}
+     *
+     * @param topics topic configuration
+     * @param kafkaProperties configuration to connect to Kafka admin tools
+     * @param groupId consumer group id to clean up
+     * @param configuration configuration for hooks that are called when running {@link #clean()}
+     * @return {@code ConsumerCleanUpRunner}
+     */
+    public static ConsumerCleanUpRunner create(@NonNull final ConsumerTopicConfig topics,
+            @NonNull final Map<String, Object> kafkaProperties,
+            @NonNull final String groupId,
+            @NonNull final ConsumerCleanUpConfiguration configuration) {
+        return new ConsumerCleanUpRunner(topics, kafkaProperties, groupId, configuration);
+    }
+
+    @Override
+    public void close() {
+        // do nothing
+    }
+
+    @Override
+    public void clean() {
+        try (final AdminClientX adminClient = this.createAdminClient()) {
+            final Task task = new Task(adminClient);
+            task.clean();
+        }
+    }
+
+    /**
+     * Reset your Consumer app by resetting consumer group offsets
+     */
+    public void reset() {
+        try (final AdminClientX adminClient = this.createAdminClient()) {
+            final ConsumerCleanUpRunner.Task task = new ConsumerCleanUpRunner.Task(adminClient);
+            task.reset();
+        }
+    }
+
+    private AdminClientX createAdminClient() {
+        return AdminClientX.create(this.kafkaProperties);
+    }
+
+    @RequiredArgsConstructor
+    private class Task {
+
+        private final @NonNull AdminClientX adminClient;
+
+        private void reset() {
+            final Optional<ConsumerGroupDescription> groupDescription =
+                    this.adminClient.consumerGroups().group(ConsumerCleanUpRunner.this.groupId).describe();
```

Comment:
You could extract a variable for `this.adminClient.consumerGroups().group(ConsumerCleanUpRunner.this.groupId)` and reuse below. Or is it possible that the group changes in between the two statements that use it?